src/js/libcs/Operators.js

```diff
@@ -5097,3 +5097,256 @@ evaluator.removeelement$1 = function(args, modifs) {
+
+        let erg1 = pslq(real);
+        let erg2 = pslq(real * real + imag * imag);
+        if (erg1 && erg2) {
+            const aa = -erg1[0] * erg2[0];
+            const bb = -erg1[1] * erg2[0] * 2;
+            const cc = erg1[0] * erg2[1];
+            return quadraticc(imag, aa, bb, cc);
+        }
+        return General.string(x);
+    }
+
+    function guess(x) {
+        if (CSNumber._helper.isAlmostZero(x))
+            return "0";
+        if (!CSNumber._helper.isAlmostReal(x))
+            return guessComplex(x);
+        const pi = Math.PI;
+        let inx = [pi,
+            x.value.real,
+            1.0
+        ];
+
+        // Run PSLQ
+        const res = PSLQ.doPSLQ(inx, 15);
+
+        let sreal = "" + x.value.real;
+        let srealquot = "";
+        let srealquad = "";
+
+        let error1 = 10000;
+        let error2 = 10000;
+        //System.err.println("TEST FRACTION");
+        let r1 = 100000;
+        if (res && (r1 = reasonable(res)) < 3000) {
+            srealquot = quotient(x.value.real, res[0], -res[1], res[2]);
+            error1 = error;
+        }
+
+
+        let inx2 = [x.value.real * x.value.real,
+            x.value.real,
+            1.0
+        ];
+        const res2 = PSLQ.doPSLQ(inx2, 15);
+        //      System.err.println("TEST QUADRATIC");
+        let r2 = 100000;
+        if (res2 && (r2 = 10 + reasonable(res2)) < 9000) {
+            srealquad = quadratic(x.value.real, res2[0], res2[1], res2[2]);
+            error2 = error;
+        }
+
+        if (r1 < r2 && error1 < 0.000000000001)
+            sreal = srealquot;
+        if (r2 < r1 && error2 < 0.000000000001)
+            sreal = srealquad;
+
+        return sreal;
+
+    }
+
+    function reasonable(res) {
+        let r = Math.abs(res[0]);
+        if (Math.abs(res[1]) > r) r = Math.abs(res[1]);
+        if (Math.abs(res[2]) > r) r = Math.abs(res[2]);
+        return r;
+    }
+
+    /**
+     * Find a fractional representation of the quotient of two numbers.
+     *
+     * @param a numerator
+     * @param b divisor
+     * @return a string that represents the quotient and is fully canceled.
+     *         The sign is included (+ or -)
+     */
+    function fraction(a, b) {
+        let str = "+";
+        if (a * b < 0) {
+            a = Math.abs(a);
+            b = Math.abs(b);
+            str = "-";
+        }
+        const r = gcd(a, b);
+        a /= r;
+        b /= r;
+        if (a < 0 && b < 0) {
+            a = -a;
+            b = -b;
+        }
+        if (b === 1)
+            str = str + a;
+        else
+            str = str + a + "/" + b;
+        return str;
+    }
+
+    /**
+     * @param a an integer
+     * @return a square divisor of a
+     */
+    function extractsquare(a) {
+        for (let n = 3000; n > 0; n--)
+            if (a % (n * n) === 0)
+                return n;
+        return 1;
+    }
+
+
+    /**
+     * @return a string representation of a complex number
```

Comment:
There must be an error somewhere here because `guess(i*pi)` gives `NaN/NaN-i*1/NaN*sqrt(NaN)`